library/aes.c

```diff
@@ -983,6 +1042,163 @@ int mbedtls_aes_crypt_cbc( mbedtls_aes_context *ctx,
 }
 #endif /* MBEDTLS_CIPHER_MODE_CBC */
 
+#if defined(MBEDTLS_CIPHER_MODE_XTS)
+/*
+ * AES-XEX buffer encryption/decryption
+ */
+int mbedtls_aes_crypt_xex( mbedtls_aes_xts_context *ctx,
+                           int mode,
+                           size_t length,
+                           const unsigned char iv[16],
+                           const unsigned char *input,
+                           unsigned char *output )
+{
+    if( length % 16 )
```

Comment:
I don't think we always compare with zero explicitly. Consider `mbedtls_aes_crypt_cbc()` for example.

I will compare with zero in the new code, since there seems to be the impression that we always compare with zero explicitly.

Fixed